pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -219,20 +230,89 @@ func (a ClusterOperatorAnonymizer) Marshal(_ context.Context) ([]byte, error) {
 	return runtime.Encode(serializer, a.ClusterOperator)
 }
 
-// type ClusterOperatorListAnonymizer struct{ *configv1.ClusterOperatorList }
+func isFailingOperator(operator *configv1.ClusterOperator) bool {
+	for _, condition := range operator.Status.Conditions {
+		switch {
+		case condition.Type == configv1.OperatorDegraded && condition.Status == configv1.ConditionTrue,
+			condition.Type == configv1.OperatorAvailable && condition.Status == configv1.ConditionFalse:
+			return true
+		}
+	}
+	return false
+}
+
+func namespacesForOperator(operator *configv1.ClusterOperator) []string {
+	var ns []string
+	for _, ref := range operator.Status.RelatedObjects {
+		if ref.Resource == "namespaces" {
+			ns = append(ns, ref.Name)
+		}
+	}
+	return ns
+}
 
-// func (a ClusterOperatorListAnonymizer) Marshal(_ context.Context) ([]byte, error) {
-// 	return runtime.Encode(serializer, a.ClusterOperatorList)
-// }
+type NodeAnonymizer struct{ *corev1.Node }
 
-// func clusterOperatorListResourceVersion(items *configv1.ClusterOperatorList) string {
-// 	rvs := make([]string, 0, len(items.Items))
-// 	for _, item := range items.Items {
-// 		rvs = append(rvs, item.ResourceVersion)
-// 	}
-// 	sort.Strings(rvs)
-// 	return strings.Join(rvs, ",")
-// }
+func (a NodeAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	return runtime.Encode(kubeSerializer, anonymizeNode(a.Node))
+}
+
+func anonymizeNode(node *corev1.Node) *corev1.Node {
+	for k := range node.Annotations {
+		node.Annotations[k] = ""
+	}
+	for k, v := range node.Labels {
+		node.Labels[k] = strings.Repeat("x", len(v))
+	}
+	node.Status.Addresses = nil
+	node.Status.DaemonEndpoints = corev1.NodeDaemonEndpoints{}
+	node.Status.NodeInfo.BootID = strings.Repeat("x", len(node.Status.NodeInfo.BootID))
+	node.Status.NodeInfo.SystemUUID = strings.Repeat("x", len(node.Status.NodeInfo.SystemUUID))
+	node.Status.NodeInfo.MachineID = strings.Repeat("x", len(node.Status.NodeInfo.MachineID))
+	node.Status.Images = nil
+	return node
+}
+
+func isFailingNode(node *corev1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionFalse {
+			return true
+		}
+	}
+	return false
+}
+
+type PodAnonymizer struct{ *corev1.Pod }
+
+func (a PodAnonymizer) Marshal(_ context.Context) ([]byte, error) {
+	return runtime.Encode(kubeSerializer, anonymizePod(a.Pod))
+}
+
+func anonymizePod(pod *corev1.Pod) *corev1.Pod {
+	// pods gathered from openshift namespaces and cluster operators are expected to be under our control and contain
+	// no sensitive information
+	return pod
+}
+
+func isFailingPod(pod *corev1.Pod) bool {
```

Comment:
do we consider Pending pods as failing if the operator report degraded? In some cases (like DS pods), the operator  report unavailable or degraded because it does not see pods rolled out. The reason why the pod is not running might be useful to know. (in most cases it is failed node, but it might be bad image or something)

Reply:
ok.